lib/feature_flags.rb

```diff
@@ -54,23 +54,21 @@ def city_features(city_id: , feature_list: [])
     features
   end
 
-  def activate_user(feature:, city_id:, id:)
-    if city_live?(feature: feature, city_id: city_id)
+  def activate_user(feature:, id:, live: false)
+    if live
       @redis.srem(blacklist_user_key(feature, id), id)
-    elsif city_beta?(feature: feature, city_id: city_id)
+      @redis.srem(whitelist_user_key(feature), id)
+    else
       @redis.sadd(whitelist_user_key(feature), id)
```

Comment:
You mean `@redis.srem(whitelist_user_key(feature), id)` this line?

I remove it from the whitelist here because the `user_state` checks whitelist and return `beta` first before it checks the blacklist.